### PR TITLE
Make sure all the tests also pass on Windows OS

### PR DESCRIPTION
**Pull request by Diogomrol**
Some tests (2) were only working in Linux OS for this plugin. I made a
change to make them pass also on Windows OS by checking the OS. Maybe this can be improved, but it's a first take.

### CODE REVIEW

File: src/test/java/cd/go/artifact/docker/registry/model/BuildFileArtifactPlanConfigTest.java [hunk 1]
```diff
@@ -89,7 +90,13 @@ public void shouldErrorOutWhenFileDoesNotExist() throws UnresolvedPropertyExcept
         final ArtifactPlanConfig artifactPlanConfig = new BuildFileArtifactPlanConfig("random.json");
 
         thrown.expect(RuntimeException.class);
-        thrown.expectMessage(String.format("%s/random.json (No such file or directory)", agentWorkingDir.getAbsolutePath()));
+        if (SystemUtils.IS_OS_WINDOWS) {
+            thrown.expectMessage(String.format("%s\\random.json (The system cannot find the file specified)", agentWorkingDir.getAbsolutePath()));
```
Comment (ketan) [inline src/test/java/cd/go/artifact/docker/registry/model/BuildFileArtifactPlanConfigTest.java]: Use `File.pathSeparator` perhaps?

```
throw.expectMessage("%s%s/random.json....", agentWorkingDir.getAbsolutePath(), File.pathSeparator)
```

Comment (Diogomrol) [reply src/test/java/cd/go/artifact/docker/registry/model/BuildFileArtifactPlanConfigTest.java]: You mean File.separator maybe, right? pathSeparator will be ';' on Windows, e.g.

Comment (ketan) [reply src/test/java/cd/go/artifact/docker/registry/model/BuildFileArtifactPlanConfigTest.java]: Yes `File.separator`. Sorry, I always get confused between them.

Comment (Diogomrol) [reply src/test/java/cd/go/artifact/docker/registry/model/BuildFileArtifactPlanConfigTest.java]: Committed a new version :)

**Comment by varshavaradarajan** [comment]
If the BuildFileArtifactPlanConfigTest wasn't passing on Windows, then the build file behavior must be checked on Windows in addition to fixing the test. cc @bdpiparva, @akshaydewan

@Diogomrol - thank you for the attempting the S3 plugin. Usually, each plugin has it's own repository though. May be, you could create a new s3 artifact plugin repository?

**Comment by Diogomrol** [comment]
The build file seems to be working fine here :)

About the repository, I forked this to start but you are right, it makes sense. I will create a repo still today for that one. Thanks!

**Comment by Diogomrol** [comment]
That helps a lot :) Thank you for the help, I will move with creating the repo and hopefully soon the plugin will be working well :)

Thanks!

**Comment by bdpiparva** [comment]
Closing the PR as there is no activity. Please feel free to reopen it if you are still working on it.

**Pull request closed without merging**